src/Icons/src/IconRegistryInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons;
+
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @extends \IteratorAggregate<string>
+ *
+ * @internal
+ */
+interface IconRegistryInterface extends \IteratorAggregate, \Countable
+{
+    /**
+     * @return array{0: string, 1: array<string, string|bool>}
+     *
+     * @throws IconNotFoundException
+     */
+    public function get(string $name): array;
```

Comment:
Made the PR against your other branch, so i'll write here too: if we expose an Interface, we definitively should use a POO instead of an array

Reply:
> if we expose an Interface

Technically, this interface isn't exposed (it's internal)...